src/tests/cptests/cptests.cc

```diff
@@ -951,6 +976,57 @@ void cptest_servicestatus()
     delete cc;
 }
 
+void cptest_sendsignal()
+{
+    using namespace std;
+
+    service_set sset;
+    ha_string command = "test-command";
+    list<pair<unsigned,unsigned>> command_offsets;
+    command_offsets.emplace_back(0, command.length());
+    std::list<prelim_dep> depends;
+
+    process_service p {&sset, "test-service", std::move(command), command_offsets, depends};
+    init_service_defaults(p);
+    sset.add_service(&p);
+
+    p.start();
+    sset.process_queues();
+    base_process_service_test::exec_succeeded(&p);
+    sset.process_queues();
+
+    int fd = bp_sys::allocfd();
+    auto *cc = new control_conn_t(event_loop, &sset, fd);
+
+    // Get a service handle:
+    control_conn_t::handle_t h = find_service(fd, "test-service", service_state_t::STARTED,
+            service_state_t::STARTED);
+
+    // Prepare a signal: (SIGHUP for example)
+    uint32_t sig = SIGHUP;
+
+    // Issue a signal:
+    std::vector<char> cmd = { DINIT_CP_SIGNAL };
+    char * sig_cp = reinterpret_cast<char *>(&sig);
+    char * h_cp = reinterpret_cast<char *>(&h);
+    cmd.insert(cmd.end(), sig_cp, sig_cp + sizeof(sig));
+    cmd.insert(cmd.end(), h_cp, h_cp + sizeof(h));
+
+    bp_sys::supply_read_data(fd, std::move(cmd));
+    event_loop.regd_bidi_watchers[fd]->read_ready(event_loop, fd);
+
+    std::vector<char> wdata;
+    bp_sys::extract_written_data(fd, wdata);
+    assert(wdata.size() == 1);
+    assert(wdata[0] == DINIT_RP_ACK);
+
+    assert(bp_sys::last_sig_sent == SIGHUP);
```

Comment:
This is basically a good idea for a test, but it is weak because it only checks one signal. The test could pass just because `bp_sys::last_sig_sent` happened to already be `SIGHUP` even if this wasn't changed at all by the control message. Ideally, it should now also send another DINIT_CP_SIGNAL now with a different signal number and check that `bp_sys::last_sig_sent` changes.